### ISSUE TITLE
AWC camel cases headers with more than one hyphen incorrectly

### ISSUE THREAD

**Issue by ryanbai1412** [actix/actix-web#2674]
## Expected Behavior
The headers should be camel cased correctly

## Current Behavior
The headers are not camel cased correctly (`A-B-C` becomes `A-BCc`, `A-B-C-D` becomes `A-BCD-d`).

## Possible Solution
https://github.com/actix/actix-web/blob/e7a05f98925dcd8461845b67fa5769b24aa88961/actix-http/src/h1/encoder.rs#L514-L523
At line 518 I believe `index` should be incremented because the iterator was advanced.

## Steps to Reproduce (for bugs)
```rs
#[tokio::main]
async fn main() {
    let client = awc::Client::new();
    let body = client
        .get("http://httpbin.org/get")
        .camel_case()
        .insert_header(("A-B-C", ""))
        .insert_header(("A-B-C-D", ""))
        .send()
        .await
        .unwrap()
        .body()
        .await
        .unwrap();
    println!("{}", String::from_utf8_lossy(&body));
}
```

## Your Environment

- Rust Version (I.e, output of `rustc -V`): `rustc 1.59.0 (9d1b2106e 2022-02-23)`
- Actix Web Version: `awc 3.0.0-beta.21`

**Comment by woile** [comment]
I'm gonna try to work on this, just to be clear, the correct ways should be:

A-B-C becomes A-B-C, 
A-B-C-D becomes A-B-C-D

right? so, they remain the same as the original values

**Comment by robjtede** [comment]
Released in `actix-http` v3.0.2. `cargo update` to get it.
